search-relevancy-test/src/search_relevancy_test/edsc_log_parser.clj

```diff
@@ -0,0 +1,123 @@
+  [line]
+  (try
+    (-> (re-find #"\{.*\}" line)
+        (json/parse-string true)
+        :data)
+    (catch Exception e
+      nil)))
+
+(defn create-test-from-line
+  "Returns an anomaly test (as a map) from the provided log line. Returns nil if no test should be
+  created from the log line."
+  [line test-number]
+  (let [metrics (get-relevancy-json-from-line line)]
+    (when (pos? (get-selected-index metrics))
+      (let [selected-collection (get-selected-collection metrics)
+            top-collection (get-top-collection metrics)
+            cmr-search (get-cmr-search-from-metric metrics)]
+        (when cmr-search
+          {:search cmr-search
+           :concept-ids [selected-collection top-collection]
+           :test-number test-number})))))
+
+(defn get-all-tests
+  "Returns all of the tests for the given log-file"
+  [log-file]
+  (let [counter (atom 0)
+        reader (io/reader log-file)]
+    (for [line (line-seq reader)
+          :let [test (create-test-from-line line (swap! counter inc))]
+          :when test]
+      test)))
+
+(defn get-contradicting-tests
+  "Returns any tests where there are two tests that say contradictory things (note that sense we
+  are currently ignoring anything with a selected_index of 0 we can't have any contradictions.)"
+  [tests]
+  nil)
+
+(defn remove-duplicate-tests
+  "Returns a set of relevancy tests with any duplicates removed while still preserving the test
+  order."
+  [tests]
+  (->> (for [test tests]
+        [(select-keys test [:search :concept-ids]) test])
+       (into {})
+       vals
+       (sort-by :test-number)))
+
+(defn- test->anomaly-row
+  "Returns a row for the anomaly CSV file "
+  [test n]
+  [n
+   1
+   (:search test)
+   (string/join "," (:concept-ids test))
+   (str "EDSC user anomaly line " (:test-number test))])
+
+(def anomaly-csv-columns
+  "Columns in the anomaly test file."
+  ["Anomaly","Test","Search","Concept-ids","Description"])
+
+(defn write-edsc-anomaly-csv
+  "Writes out the tests to an EDSC anomaly CSV file"
+  [tests output-filename]
+  (let [counter (atom 0)
+        row-data (for [test tests]
+                   (test->anomaly-row test (swap! counter inc)))]
+   (with-open [csv-file (io/writer output-filename)]
+     (csv/write-csv csv-file [anomaly-csv-columns])
+     (csv/write-csv csv-file row-data))))
+
+(def anomaly-filename
+  "File to save EDSC anomaly tests to."
+  "edsc_anomaly_tests.csv")
+
+(defn parse-edsc-logs
+  "Parses the provided EDSC relevancy log file and returns search relevancy tests."
+  [log-file]
+  (let [all-tests (remove-duplicate-tests (get-all-tests log-file))]
+    (write-edsc-anomaly-csv all-tests (str "resources/" anomaly-filename))
+    all-tests))
+
+(comment
+ (count (get-all-tests (io/resource "EDSC_Relevancy_metrics.log")))
+ (count (remove-duplicate-tests (get-all-tests (io/resource "EDSC_Relevancy_metrics.log"))))
+ (parse-edsc-logs (io/resource "one-ignored-request.log"))
```

Comment:
Is this section debug code or is it to run the tests manually?  If it is debug code, shouldn't we get rid of it?  You could create test functions to do the manual runs.

Reply:
Removed debug.